src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -358,6 +358,10 @@ private function createFirewall(ContainerBuilder $container, $id, $firewall, &$a
             }
             $defaultProvider = $providerIds[$normalizedName];
         } else {
+            if (count($providerIds) > 1) {
+                @trigger_error(sprintf('Not configuring explicitly the provider on "%s" firewall is ambiguous as there is more than one registered provider. Using the first configured provider is deprecated since 3.4 and it will throw an exception on 4.0 instead.', $id), E_USER_DEPRECATED);
```

Comment:
I suggest

> Firewall "$id" has no "provider" set but multiple providers exist. Using the first configured provider ($defaultProvider) is deprecated since 3.4 and will throw an exception in 4.0, set the "provider" key on the firewall instead.

Reply:
Perfect, thanks 👍